app/code/Magento/PageCache/Model/HTTP/Header/XMagentoTags.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Magento\PageCache\Model\HTTP\Header;
+
+use Zend\Http\Header\MultipleHeaderInterface;
+use Zend\Http\Header\GenericHeader;
+use Zend\Http\Header\HeaderValue;
+use Zend\Http\Header\Exception;
+
+class XMagentoTags implements MultipleHeaderInterface
+{
+    /**
+     * @var string
+     */
+    protected $value;
+
+    /**
+     * XMagentoTags constructor.
+     *
+     * @param string|null $value
+     */
+    public function __construct($value = null)
+    {
+        if ($value) {
+            HeaderValue::assertValid($value);
+            $this->value = $value;
+        }
+    }
+
+    /**
+     * Create X-Magento-Tags header from a given header line
+     *
+     * @param string $headerLine The header line to parse.
+     *
+     * @return self
+     * @throws Exception\InvalidArgumentException If the name field in the given header line does not match.
+     */
+    public static function fromString($headerLine)
+    {
+        list($name, $value) = GenericHeader::splitHeaderLine($headerLine);
+
+        // check to ensure proper header type for this factory
+        if (strtolower($name) !== 'x-magento-tags') {
+            throw new Exception\InvalidArgumentException(
+                sprintf(
+                    'Invalid header line for X-Magento-Tags string: "%s"',
+                    $name
+                )
+            );
+        }
+
+        // @todo implementation details
+        $header = new static($value);
+
+        return $header;
+    }
+
+    /**
+     * Cast multiple header objects to a single string header
+     *
+     * @param  array $headers
+     * @throws Exception\InvalidArgumentException
+     * @return string
+     */
+    public function toStringMultipleHeaders(array $headers)
```

Comment:
This method seems to be copied from the Zend's default implementation of `\Zend\Http\Header\GenericMultiHeader::toStringMultipleHeaders` which could make it more challenging to support in the future